psydac/cad/geometry.py

```diff
@@ -85,7 +85,7 @@ def __init__(self, domain=None, ncells=None, periodic=None, mappings=None,
                 assert isinstance(periodic, dict)
 
             # ... check sanity
-            interior_names = domain.interior_names
+            interior_names = sorted(domain.interior_names) if domain.logical_domain is None else sorted(domain.logical_domain.interior_names)
```

Comment:
I think here lies the issue for the currently failing tests. Why do we want to use the `domain.logical_domain.interior_names` instead of the `domain.interior_names`? The problem is that the `ncells` dictionary has `domain.interior_names` as keys.

Reply:
This bug was definitely introduced by the "optimistic" merge with `devel` that we did yesterday. Have you tried to run the tests locally with the change you propose?

Reply:
this was fixed by these changes [4a07ae6](https://github.com/pyccel/psydac/pull/241/commits/4a07ae6a8ff2ec7b4c9471b7079d838c3cbb0944)

Reply:
Excellent, thanks!